sympy/polys/rootoftools.py

```diff
@@ -1166,7 +1178,9 @@ def __new__(cls, expr, func=None, x=None, auto=True, quadratic=False):
         func = Lambda(var, expr)
 
         rational = cls._is_func_rational(poly, func)
-        factors, terms = _pure_factors(poly), []
+        _, factors = poly.factor_list()
+        factors = [(PurePoly(f, expand=False), m) for f, m in factors]
```

Comment:
Why is this using PurePoly?

Reply:
It might make sense to handle RootSum in a separate PR but ultimately we should not use PurePoly for RootSum either.